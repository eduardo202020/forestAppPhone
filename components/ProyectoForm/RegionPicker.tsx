import React from 'react';
import { Picker } from '@react-native-picker/picker';
import { Text, StyleSheet } from 'react-native';

interface RegionPickerProps {
  value: string;
  onChange: (v: string) => void;
  editable?: boolean;
  error?: string;
}

const RegionPicker: React.FC<RegionPickerProps> = ({
  value,
  onChange,
  editable = true,
  error,
}) => (
  <>
    <Picker
      selectedValue={value}
      onValueChange={onChange}
      style={styles.input}
      enabled={editable}
    >
      <Picker.Item label="Selecciona una región" value="" />
      <Picker.Item label="Loreto" value="Loreto" />
      <Picker.Item label="Cusco" value="Cusco" />
      <Picker.Item label="Puno" value="Puno" />
      <Picker.Item label="Madre de Dios" value="Madre de Dios" />
      <Picker.Item label="Amazonas" value="Amazonas" />
      <Picker.Item label="San Martín" value="San Martín" />
    </Picker>
    {error && <Text style={styles.errorText}>{error}</Text>}
  </>
);

const styles = StyleSheet.create({
  input: {
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 8,
    padding: 10,
    marginBottom: 12,
    fontSize: 16,
    backgroundColor: '#f9f9f9',
  },
  errorText: {
    color: 'red',
    marginBottom: 8,
    fontSize: 13,
  },
});

export default RegionPicker;
