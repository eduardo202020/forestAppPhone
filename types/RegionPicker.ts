export interface RegionPickerProps {
  value: string;
  onChange: (value: string) => void;
  editable?: boolean;
  error?: string;
}
