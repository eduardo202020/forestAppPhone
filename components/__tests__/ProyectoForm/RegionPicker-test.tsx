import React from 'react';
import renderer from 'react-test-renderer';
import RegionPicker from '../../ProyectoForm/RegionPicker';

describe('RegionPicker', () => {
  it('se renderiza sin errores', () => {
    const tree = renderer
      .create(
        <RegionPicker value="" onChange={() => {}} editable error={undefined} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('muestra el error si existe', () => {
    const tree = renderer
      .create(
        <RegionPicker
          value=""
          onChange={() => {}}
          editable
          error="Campo requerido"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
