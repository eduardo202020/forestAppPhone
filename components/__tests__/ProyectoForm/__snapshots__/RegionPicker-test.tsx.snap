// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegionPicker muestra el error si existe 1`] = `
[
  <Text
    style={
      {
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Región
  </Text>,
  <View
    style={
      {
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": 10,
      }
    }
  >
    <RNCPicker
      items={
        [
          {
            "label": "Selecciona una región",
            "testID": undefined,
            "textColor": undefined,
            "value": "",
          },
          {
            "label": "Loreto",
            "testID": undefined,
            "textColor": undefined,
            "value": "Loreto",
          },
          {
            "label": "Cusco",
            "testID": undefined,
            "textColor": undefined,
            "value": "Cusco",
          },
          {
            "label": "Puno",
            "testID": undefined,
            "textColor": undefined,
            "value": "Puno",
          },
          {
            "label": "Madre de Dios",
            "testID": undefined,
            "textColor": undefined,
            "value": "Madre de Dios",
          },
          {
            "label": "Amazonas",
            "testID": undefined,
            "textColor": undefined,
            "value": "Amazonas",
          },
          {
            "label": "San Martín",
            "testID": undefined,
            "textColor": undefined,
            "value": "San Martín",
          },
        ]
      }
      numberOfLines={1}
      onChange={[Function]}
      selectedIndex={0}
      style={
        [
          {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>,
  <Text
    style={
      {
        "color": "red",
        "fontSize": 12,
      }
    }
  >
    Campo requerido
  </Text>,
]
`;

exports[`RegionPicker se renderiza sin errores 1`] = `
[
  <Text
    style={
      {
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Región
  </Text>,
  <View
    style={
      {
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": 10,
      }
    }
  >
    <RNCPicker
      items={
        [
          {
            "label": "Selecciona una región",
            "testID": undefined,
            "textColor": undefined,
            "value": "",
          },
          {
            "label": "Loreto",
            "testID": undefined,
            "textColor": undefined,
            "value": "Loreto",
          },
          {
            "label": "Cusco",
            "testID": undefined,
            "textColor": undefined,
            "value": "Cusco",
          },
          {
            "label": "Puno",
            "testID": undefined,
            "textColor": undefined,
            "value": "Puno",
          },
          {
            "label": "Madre de Dios",
            "testID": undefined,
            "textColor": undefined,
            "value": "Madre de Dios",
          },
          {
            "label": "Amazonas",
            "testID": undefined,
            "textColor": undefined,
            "value": "Amazonas",
          },
          {
            "label": "San Martín",
            "testID": undefined,
            "textColor": undefined,
            "value": "San Martín",
          },
        ]
      }
      numberOfLines={1}
      onChange={[Function]}
      selectedIndex={0}
      style={
        [
          {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>,
]
`;
